mmv1/third_party/terraform/provider/provider_internal_test.go

```diff
@@ -688,3 +688,307 @@ func TestProvider_providerConfigure_project(t *testing.T) {
+			setupTestEnvs(t, tc.EnvVariables)
+			p := Provider()
+			d := tpgresource.SetupTestResourceDataFromConfigMap(t, p.Schema, tc.ConfigValues)
+
+			// Act
+			c, diags := providerConfigure(ctx, d, p)
+
+			// Assert
+			if diags.HasError() && !tc.ExpectError {
+				t.Fatalf("unexpected error(s): %#v", diags)
+			}
+			if !diags.HasError() && tc.ExpectError {
+				t.Fatal("expected error(s) but got none")
+			}
+			if diags.HasError() && tc.ExpectError {
+				v, ok := d.GetOk("request_reason")
+				if ok {
+					val := v.(string)
+					if val != tc.ExpectedSchemaValue {
+						t.Fatalf("expected request_reason value set in provider data to be %s, got %s", tc.ExpectedSchemaValue, val)
+					}
+					if tc.ExpectFieldUnset {
+						t.Fatalf("expected request_reason value to not be set in provider data, got %s", val)
+					}
+				}
+				// Return early in tests where errors expected
+				return
+			}
+
+			v := d.Get("request_reason")
+			val := v.(string)
+			config := c.(*transport_tpg.Config) // Should be non-nil value, as test cases reaching this point experienced no errors
+
+			if v != tc.ExpectedSchemaValue {
+				t.Fatalf("expected request_reason value set in provider data to be %s, got %s", tc.ExpectedSchemaValue, val)
+			}
+			if config.RequestReason != tc.ExpectedConfigValue {
+				t.Fatalf("expected request_reason value in provider struct to be %s, got %s", tc.ExpectedConfigValue, config.Credentials)
+			}
+		})
+	}
+}
+
+func TestProvider_providerConfigure_batching(t *testing.T) {
+	//var batch []interface{}
+	cases := map[string]struct {
+		ConfigValues                map[string]interface{}
+		EnvVariables                map[string]string
+		ExpectError                 bool
+		ExpectFieldUnset            bool
+		ExpectedEnableBatchingValue bool
+		ExpectedSendAfterValue      string
+	}{
+		"if batch is an empty block, it will set the default values": {
+			// Although at the schema level it's shown that by default it's set to false, the actual default value
+			// is true and can be seen in the `ExpanderProviderBatchingConfig` struct
+			// https://github.com/GoogleCloudPlatform/magic-modules/blob/8cd4a506f0ac4db7b07a8cce914449d34df6f20b/mmv1/third_party/terraform/transport/config.go.erb#L504-L508
+			ExpectedEnableBatchingValue: false,
+			ExpectedSendAfterValue:      "", // uses "" value to be able to set the default value of 30s
+			ExpectFieldUnset:            true,
+		},
+		"if batch is configured with both enable_batching and send_after": {
+			ConfigValues: map[string]interface{}{
+				"batching": []interface{}{
+					map[string]interface{}{
+						"enable_batching": true,
+						"send_after":      "10s",
+					},
+				},
+			},
+			ExpectedEnableBatchingValue: true,
+			ExpectedSendAfterValue:      "10s",
+		},
+		"if batch is configured with only enable_batching": {
+			ConfigValues: map[string]interface{}{
+				"batching": []interface{}{
+					map[string]interface{}{
+						"enable_batching": true,
+					},
+				},
+			},
```

Comment:
If you check [this output](https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/5870779051/job/15918512238) from checks running unit tests from this PR it complains about lacking authentication:

```
    --- FAIL: TestProvider_providerConfigure_batching/if_batch_is_configured_with_only_enable_batching (0.03s)
        provider_internal_test.go:952: unexpected error(s): diag.Diagnostics{diag.Diagnostic{Severity:0, Summary:"Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded. To use your gcloud credentials, run 'gcloud auth application-default login'.  Original error: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information", Detail:"", AttributePath:cty.Path(nil)}}
```

a way to solve this is to supply these fake creds in the `ConfigValues` values for test cases:

```diff
			ConfigValues: map[string]interface{}{
+   			        "credentials":  transport_tpg.TestFakeCredentialsPath,
				"batching": []interface{}{
					map[string]interface{}{
						"enable_batching": true,
					},
				},
			},
```

Reply:
This is something I've done in equivalent tests elsewhere, e.g. see: https://github.com/GoogleCloudPlatform/magic-modules/pull/7723/files#diff-89e3dcb3dbfe5488060c297a3f4dd633f603f25ca824385b926cd9bec5980d39